core/payment/src/service.rs

```diff
@@ -146,7 +146,7 @@ mod public {
         };
 
         let offeror_id = agreement.offer.provider_id.clone().unwrap(); // FIXME: provider_id shouldn't be an Option
-        let issuer_id = debit_note.issuer_id.clone();
+        let issuer_id = debit_note.issuer_id.to_string();
```

Comment:
id rather parse sender_id here and compare node_ids below

Reply:
But parsing requires error handling and this doesn't. Also, I would have to parse offeror_id as as well. So how is that better?

Reply:
it was a suggestion / non-blocking comment :)

Reply:
OK :)